src/x86/x86_model.py

```diff
@@ -492,25 +492,31 @@ def rollback(self) -> int:
         self.dependencies = self.dependency_checkpoints.pop()
         return super().rollback()
 
+
 class X86UnicornVSPECUnknown(X86FaultModelAbstract):
     """
     Contract for value speculation with unknown values
     """
     reg_taints: Dict
     flag_taints: Dict
-    address_taints: Dict
     reg_taints_checkpoints: List[Dict]
-    curr_observation: set()
+    curr_observation : Set = set()
 
     def __init__(self, *args):
         super().__init__(*args)
         # DIV exceptions only for now
         self.relevant_faults.update([21])
         self.reg_taints = dict()
         self.flag_taints = dict()
-        self.address_taints = dict()
         self.reg_taints_checkpoints = []
 
+    def _load_input(self, input_: Input):
+        self.curr_observation.clear()
```

Comment:
The same instance of `curr_observation` is reused among runs with different inputs. We clear the set before loading the new input and we check the other sets are empty